subsys/net/ip/net_context.c

```diff
@@ -1581,6 +1581,8 @@ static int context_sendto(struct net_context *context,
 			}
 		}
 
+		net_context_set_state(context, NET_CONTEXT_CONNECTED);
```

Comment:
Why do we need to set NET_CONTEXT_CONNECTED in sendto()? So far this state was only set in TCP and in offloaded network drivers (probably also in case of TCP). I have no experience with AF_PACKET, but it doesn't sound right to set this state in sendto(). At least we don't do this in case of UDP, which is also packet/datagram based and supports sendto().

Reply:
> Why do we need to set NET_CONTEXT_CONNECTED in sendto()?

I will rework this part, I used the connected state just to make it different from listening state.